File: sdkwork-ai-sdk-csharp/Api/Models2Api.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class Models2Api
    {
        private readonly HttpClient _client;

        public Models2Api(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Retrieve model
        /// </summary>
        public async Task<OpenAiModel?> RetrieveModelAsync(string model)
        {
            return await _client.GetAsync<OpenAiModel>(ApiPaths.AiPath($"/v1/models/{model}"));
        }

        /// <summary>
        /// Delete model
        /// </summary>
        public async Task<OpenAiModelDeleteResponse?> DeleteModelAsync(string model)
        {
            return await _client.DeleteAsync<OpenAiModelDeleteResponse>(ApiPaths.AiPath($"/v1/models/{model}"));
        }

        /// <summary>
        /// List models
        /// </summary>
        public async Task<OpenAiModelList?> ListModelsAsync()
        {
            return await _client.GetAsync<OpenAiModelList>(ApiPaths.AiPath("/v1/models"));
        }
    }
}

File: sdkwork-ai-sdk-swift/Sources/API/Models2Api.swift
import Foundation

public class Models2Api {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Retrieve model
    public func retrieveModel(model: String) async throws -> OpenAiModel? {
        let response = try await client.get(ApiPaths.aiPath("/v1/models/\(model)"))
        return response as? OpenAiModel
    }

    /// Delete model
    public func deleteModel(model: String) async throws -> OpenAiModelDeleteResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/v1/models/\(model)"))
        return response as? OpenAiModelDeleteResponse
    }

    /// List models
    public func listModels() async throws -> OpenAiModelList? {
        let response = try await client.get(ApiPaths.aiPath("/v1/models"))
        return response as? OpenAiModelList
    }
}
